docs/csharp/programming-guide/generics/constraints-on-type-parameters.md

```diff
@@ -161,6 +164,50 @@ public interface IAdditionSubtraction<T> where T : IAdditionSubtraction<T>
 
 The preceding syntax would require implementers to use [explicit interface implementation](../interfaces/explicit-interface-implementation.md) for those methods. Providing the extra constraint enables the interface to define the operators in terms of the type parameters. Types that implement the interface can implicitly implement the interface methods.
 
+## Allows ref struct
+
+The `allows ref struct` anti-constraint declares that the corresponding type argument can be a [`ref struct`](../../language-reference/builtin-types/ref-struct.md) type. Instances of that type parameter must obey the following rules:
+
+- It can't be boxed.
+- It participates in [ref safety rules](~/_csharpstandard/standard/structs.md#16412-safe-context-constraint).
+- Instances can't be used where a `ref struct` type isn't allowed, such as `static` fields.
+- Instances can be marked with the `scoped` modifier.
+
+The `allows ref struct` clause isn't inherited. In the following code:
+
+```csharp
+class C<T, S>
```

Comment:
nit: I find this visually offensive 😀 what about
```suggestion
class SomeClass<T, S>
```

Reply:
I'll take the suggestion. It's an odd shorthand from working on the spec. `C`, `M` and `P` are used for a `class` *method* and *property* everywhere.